src/diagnostics_manager.js

```diff
@@ -305,23 +305,65 @@ class DiskManager {
   }
 }
 
+/**
+ * Diagnostics eMMC Manager for dpsl.diagnostics.emmc.* APIs.
+ */
+class EmmcManager {
+  /**
+   * Runs eMMC lifetime check. This routine checks the lifetime of the eMMC
+   * drive. The routine will pass if PRE_EOL_INFO is 0x01 (normal). In
+   * addition, the value of DEVICE_LIFE_TIME_EST_TYP_A and
+   * DEVICE_LIFE_TIME_EST_TYP_B will be included in the output.
+   * @return { !Promise<!Routine> }
+   * @public
+   */
+  async runEmmcLifetimeRoutine() {
+    const functionName = 'runEmmcLifetimeRoutine';
+    if (!isSupported(functionName)) {
+      throw new MethodNotFoundError(API_NAME, functionName,
+          /* chromeVersion */ 110);
+    }
+
+    return chrome.os.diagnostics.runEmmcLifetimeRoutine().then(
+        (response) => new Routine(response.id));
+  }
+}
+
 /**
  * Diagnostics NVMe Manager for dpsl.diagnostics.nmve.* APIs.
  */
 class NvmeManager {
   /**
    * Runs NVMe smartctl test.
+   * @param {dpsl.SmartctlCheckRoutineParams} params
    * @return { !Promise<!Routine> }
    * @public
    */
-  async runSmartctlCheckRoutine() {
+  async runSmartctlCheckRoutine(params = undefined) {
     const functionName = 'runSmartctlCheckRoutine';
     if (!isSupported(functionName)) {
       throw new MethodNotFoundError(API_NAME, functionName,
           /* chromeVersion */ 100);
     }
 
-    return chrome.os.diagnostics.runSmartctlCheckRoutine().then(
+    return chrome.os.diagnostics.runSmartctlCheckRoutine(params).then(
```

Comment:
will it work if we pass undefined on old versions of browser?

maybe we should call runSmartctlCheckRoutine() without any arguments if params is undefined?

Reply:
Good call, that makes sense!